File: Test/AdventureWorksFunctionalModel/Sales/ShoppingCart_Functions.cs
namespace AW.Functions;

[Named("Cart")]
public class ShoppingCart_Functions
{
    public IQueryable<ShoppingCartItem> RemoveItems(
        IQueryable<ShoppingCartItem> items,
        IContext context)
    {
        ShoppingCart_MenuFunctions.RemoveItems(items, context);
        return ShoppingCart_MenuFunctions.Cart(context);
    }

    public IContext AddToCart(Product product, IContext context) => ShoppingCart_MenuFunctions.AddToShoppingCart(product, context);

    public string? DisableAddToCart(IContext context) => ShoppingCart_MenuFunctions.DisableIfNoCustomerForUser(context);
}

File: Test/AdventureWorksLegacyModelVB/Sales/ShoppingCart_Functions.vb
Namespace AW.Functions

	<Named("Cart")>
	Public Class ShoppingCart_Functions
		Public Function RemoveItems(ByVal items As IQueryable(Of ShoppingCartItem), ByVal context As IContext) As IQueryable(Of ShoppingCartItem)
			ShoppingCart_MenuFunctions.RemoveItems(items, context)
			Return ShoppingCart_MenuFunctions.Cart(context)
		End Function

		Public Function AddToCart(ByVal product As Product, ByVal context As IContext) As IContext
			Return ShoppingCart_MenuFunctions.AddToShoppingCart(product, context)
		End Function

'INSTANT VB WARNING: Nullable reference types have no equivalent in VB:
'ORIGINAL LINE: public string? DisableAddToCart(IContext context)
		Public Function DisableAddToCart(ByVal context As IContext) As String
			Return ShoppingCart_MenuFunctions.DisableIfNoCustomerForUser(context)
		End Function
	End Class
End Namespace